app/components/base_component.rb

```diff
@@ -1,17 +1,35 @@
 class BaseComponent < ViewComponent::Base
   def before_render
-    return if @rendered_scripts
-    @rendered_scripts = true
-    if helpers.respond_to?(:enqueue_component_scripts) && self.class.scripts.present?
-      helpers.enqueue_component_scripts(*self.class.scripts)
-    end
+    render_assets unless @has_rendered_assets
   end
 
   def self.scripts
-    @scripts ||= _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+    @scripts ||= sidecar_files_basenames(['js', 'ts'])
+  end
+
+  def self.stylesheets
+    @stylesheets ||= sidecar_files_basenames(['scss'])
```

Comment:
Technically, this behavior is unused, and is a remnant of early implementation approach. That being said, it feels like it's a reasonably expected behavior to align with how `self.scripts` is implemented, if we're going to support both `self.scripts` and `self.stylesheets` regardless (even if `self.stylesheets` would be implemented by the individual component).